---
bibliography: manuscript.bib
csl: systematic-biology.csl
output:
  pdf_document:
    fig_caption: yes
    latex_engine: xelatex
---


```{r preliminaries, include=FALSE}
	# Load packages

	# General
	library( tidyverse )
	library( stringr )
	library( forcats )
  library( magrittr )
	library( digest )
	library( knitr )
	library( jsonlite )
	library( parallel )

	# Formatting
	library( xtable )

	# Graphics
	library( cowplot )
	library( ggtree )
	library( GGally )
	library( fields ) # tim.colors
	library( seriation )
	library( gridExtra )
	library( RGraphics )

	# Biological
	library( edgeR )
	library( treeio )
	library( hutan )	# https://github.com/caseywdunn/hutan
	library( ape )
	library( agalmar )	# https://github.com/caseywdunn/agalmar
	library( picante )  # cor.table
	library( phytools ) # phylosig


	# Configure knitr, see http://yihui.name/knitr/options
	opts_knit$set( progress = TRUE, verbose = TRUE )

	# Set system computational parameters
	cores = detectCores() - 1
	if ( cores < 1 ) {
		cores = 1
	}
	set.seed( 23456 )

	# Set paths to input data
	json_name = "agalma/agalma.json"

```

```{r load_data, include=FALSE, cache=TRUE}

  # Load and parse phylogenies and expression data from the
  # json file produced by agalma

	# Eventually species tree will be parsed from json, for now specify temporary tree
	# agalma.data = jsonlite::fromJSON( json_name )
	# species_tree = read.tree( text=agalma.data$speciestree )

	# For now, specify the tree as plain text
	species_tree_text = "(((((((((((((((Nanomia_bijuga:0.120563,Agalma_elegans:0.0575132)100:0.0339417,Lychnagalma_utricularia:0.0749548)100:0.0140092,(Cordagalma_sp_:0.117222,Forskalia_asymmetrica:0.0739343)100:0.0113754)100:0.0159061,(Resomia_ornicephala:0.036642,Frillagalma_vityazi:0.0604811)100:0.0190101)100:0.024866,(((Physonect_sp_:0.0317839,Marrus_claudanielis:0.0284231)100:0.0125343,Stephalia_dilata:0.0841317)82:0.00517707,Erenna_richardi:0.0414121)100:0.0124079)100:0.0161254,((((Chelophyes_appendiculata:0.0574339,Abylopsis_tetragona:0.0562432)100:0.11489,(Chuniphyes_multidentata:0.0345263,Kephyes_ovata:0.0306657)100:0.0273795)100:0.0225022,Lilyopsis_fluoracantha:0.197884)100:0.018739,((Prayidae_D27SS7:0.0600685,Prayidae_D27D2:0.0464896)100:0.00909199,Hippopodius_hippopus:0.100508)100:0.011385)100:0.0822577)100:0.0232853,((Bargmannia_elongata:0.0175209,Bargmannia_amoena:0.0201366)100:0.0187736,Bargmannia_lata:0.031329)100:0.0887499)100:0.0300558,((Apolemia_sp_:0.0291459,Apolemia_lanosa:0.0332208)95:0.00837843,Apolemia_white:0.0422333)100:0.0841266)100:0.0313749,(Physalia_physalis:0.082276,Rhizophysa_filiformis:0.105414)100:0.0750894)100:0.133951,(Hydractinia_symbiolongicarpus:0.100907,Podocoryna_carnea:0.246708)100:0.104625)99:0.0219002,Clytia_hemisphaerica:0.331138)97:0.0219604,(Hydra_magnipapillata:0.268083,Ectopleura_larynx:0.273479)100:0.11387)100:0.0827444,Aegina_citrea:0.408585)100:0.148851,(Atolla_vanhoeffeni:0.282409,Alatina_alata:0.328817):0.058436)100:0.419544,(Aiptasia_pallida:0.199088,Nematostella_vectensis:0.188379)100:0);"

	species_tree = read.tree( text=species_tree_text )

```

```{r pre_process_species_trees, include=FALSE, cache=TRUE}

	# Repartition length of edges descended from root so they are non-zero
	species_tree = hutan::slide_root_edges( species_tree )

	# species_tree = ape::root( species_tree, c( "Nematostella_vectensis", "Aiptasia_pallida" ), resolve.root=TRUE )
	# species_tree = ape::unroot(species_tree)

	stopifnot( species_tree$Nnode == length( species_tree$node.label ) )

	species_tree = ape::ladderize( species_tree )

	# Update species names
	species_tree$tip.label = sub( '_', ' ', species_tree$tip.label )

	# Make ultrametric tree
	# By default, root has a depth of 1
	species_ultrametric = chronos( species_tree, lambda=1, model="correlated", quiet=TRUE )
	class( species_ultrametric ) = "phylo"

```


# Siphonophore Phylogeny

Stefan Siebert^1,2^, Felipe Zapata^1,3^, Mark Howison^4^, Cat Munro^1^, Alejandro Damian Serrano^1^, Freya Goetz^1,5^, Phil Pugh, Steven H.D. Haddock^4^, Casey W. Dunn^1^*

^1^ Department of Ecology and Evolutionary Biology, Brown University, Providence, RI, USA

^2^ Current address: Department of Molecular & Cellular Biology, University of California at Davis, Davis, CA, USA

^3^ Current address: Department of Ecology and Evolutionary Biology, University of California Los Angeles, Los Angeles, CA, USA

^4^ Brown Data Science Practice, Brown University, Brown University, Providence, RI, USA

^5^ Current address: Smithsonian, Washington DC, USA

^4^ Monterey Bay Aquarium Research Institute, Moss Landing, CA, USA


\* Corresponding author, casey_dunn@brown.edu

## Abstract

## Introduction

Siphonophores are...


## Methods

This manuscript is an executable document computed directly from the data, providing an explicit and reproducible description of all findings. All scripts for the analyses are available in a git repository at [https://github.com/caseywdunn/siphonophore_phylogeny_2017](https://github.com/caseywdunn/siphonophore_phylogeny_2017). The most recent commit at the time of the analysis presented here was `r system("git log | head -n 1", intern=TRUE) %>% str_replace("commit ", "")`.


#### Collecting

Collection data on all examined specimens, a description of the tissue that was sampled from the colony, collection mode, sample processing details, mRNA extraction methods, sequencing library preparation methods and sequencing details are summarized in supplementary table 1. Monterey Bay and Gulf of California specimens were collected by remotely operated underwater vehicle (ROV) or during blue-water scuba dives. *Chelophyes appendiculata* and *Hippopodius hippopus* specimen were collected in the bay of Villefranche-sur-Mer, France, during a plankton trawl on 04/13/11. Available physical vouchers have been deposited at the Museum of Comparative Zoology (Harvard University), Cambridge, MA, and at the United States National Museum (Smithsonian Institution), Washington, DC. Accession numbers are given in supplementary table X. In cases where physical vouchers were unavailable we provide photographs to document species identity (table x).


#### mRNA isolation and RNAseq library preparation

When possible specimens were starved overnight in filtered seawater at temperatures close to ambient water temperatures at the time point of specimen collection (supplementary table 1). mRNA was extracted directly from tissue using Magnetic mRNA Isolation Kit (NEB, #S1550S), Invitrogen Dynabeads mRNA Direct Kit (Ambion, #61011) or from total RNA after Trizol (Ambion, #15596026) extraction and through purification using Dynabeads mRNA Purification Kit (Ambion, #61006). Extractions were performed according to the manufacturer's instruction. In case of very small total RNA quantities, only a single round of bead purification was performed (supplementary table 1). Resulting higher rRNA read counts were dealt with further downstream in the bioinformatics workflow. Libraries were prepped for sequencing using the Illumina TruSeq RNA Sample Prep Kit (Illumina, #FC-122-1001, #FC-122-1002), the Illumina TruSeq Stranded Library Prep Kit (Illumina, #RS-122-2101) or the NEBNext RNA Sample Prep. Master Mix Set (NEB, #E6110S). We collected long read paired end Illumina data for *de novo* transcriptome assembly. Summary statistics for expression libaries are given in Table 1.


#### Analysis

Sequence assembly, annotation, phylogenetic analysis, and derivation of expression counts were conducted with the tool Agalma [@Dunn:2013kw], v0.XXX. Subsequent analyses were conducted in R and integrated into this manuscript with the `knitr` package. See Supplementary Information for R package version numbers.







## Results and Discussion

### Sample collecting and sequencing

```{r summary_table, results='asis', cache=TRUE, echo=FALSE, comment=NA, warning=FALSE}



```
Table 1 (at the end of file): Summary statistics for libraries.


### Species phylogeny

```{r species_phylogram, cache=TRUE, echo=FALSE, comment=NA, warning=FALSE, fig.cap="Phylogram of siphonophore relationships. Node labels indicate bootstrap support percent, unnumbered nodes have 100% support. The image was rendered with ggtree [@Yu:2016fo]" }

	siphonophora_node = ape::mrca( species_tree )[ "Physalia physalis", "Agalma elegans" ]
	cystonectae_node = ape::mrca( species_tree )[ "Physalia physalis", "Rhizophysa filiformis" ]
	calycophora_node = ape::mrca( species_tree )[ "Kephyes ovata", "Hippopodius hippopus" ]
	euphysonectae_node = ape::mrca( species_tree )[ "Agalma elegans", "Erenna richardi" ]

	support = as.numeric( species_tree$node.label )
	support[ support > 99 ] = NA

	# pad tip values
	support = c( rep( NA, length( species_tree$tip.label ) ), support )

	p = ggtree( species_tree ) +
			geom_text( aes( label=support ), vjust=-.5, hjust=1, size=2.5, col="darkslategray4" ) +
			geom_tiplab( size=3 ) +
			xlim( NA, 4 ) +
			geom_cladelabel( node=cystonectae_node, label="Cystonectae", align=FALSE, offset=1, color='red' ) +
			geom_cladelabel( node=calycophora_node, label="Calycophora", align=FALSE, offset=1, color='blue' ) +
			geom_strip(
				which( species_tree$tip.label=="Agalma elegans" ),
				which( species_tree$tip.label=="Apolemia white" ),
				barsize=0.5,
				color='orange',
				offset=1,
				label='"Physonectae"'
			) +
			geom_strip(
				which( species_tree$tip.label=="Hydractinia symbiolongicarpus" ),
				which( species_tree$tip.label=="Nematostella vectensis" ),
				barsize=0.5,
				color='gray',
				offset=1,
				label='"Outgroups"'
			)

	flip( p, euphysonectae_node, calycophora_node )

```


The phylogenetic relationships between siphonophores species presented here is based on XXX genes from XXX siphonophore species, including new data presented here for XXX species. These findings are entirely consistent with a previous analysis based on two genes (16S and 18S ribosomal RNA) [@Dunn:2005dy]. Cystonectae is the sister group to the remaining siphonophores, and Calycophorae is nested within the paraphyletic "Physonectae". In addition, multiple nodes that were not resolved in the previous two-gene analysis do recieve strong support in this XXX-gene transcriptome analysis. These findings include XXX.



## Conclusion



## Acknowledgements

This work was supported by the National Science Foundation (DEB-1256695 and the Waterman Award). Sequencing at the Brown Genomics Core facility was supported in part by NIH P30RR031153 and NSF EPSCoR EPS-1004057. Data transfer was supported by NSF RII-C2 EPS-1005789. Analyses were conducted with computational resources and services at the Center for Computation and Visualization at Brown University, supported in part by the NSF EPSCoR EPS-1004057 and the State of Rhode Island. We also thank the MBARI crews and ROV pilots for collection of the specimens.



## Supplementary Information



### Supplementary Analyses



### Software versions

This manuscript was computed on `r format( Sys.time(), "%a %b %d %X %Y" )` with the following R package versions.

```{r session_summary, echo=FALSE, comment=NA}
	sessionInfo()
```


## References
